src/app/modules/shared_models/message_reaction_item.nim

```diff
@@ -0,0 +1,88 @@
+import json, strformat
+
+type
+  EmojiId* {.pure.} = enum
+    Heart = 1,
+    Thumbsup,
+    Thumbsdown,
+    Laughing,
+    Cry,
+    Angry
+
+type
+  ReactionDetails* = object
+    publicKey: string
+    displayName: string
+    reactionId: string
+
+type 
+  MessageReactionItem* = object
+    emojiId: EmojiId
+    amIReactedWithThisEmoji: bool
+    reactions: seq[ReactionDetails]
+
+proc toEmojiIdAsEnum*(emojiId: int, emojiIdAsEnum: var EmojiId): bool = 
+  if(emojiId >= ord(low(EmojiId)) or emojiId <= ord(high(EmojiId))): 
+    emojiIdAsEnum = EmojiId(emojiId)
+    return true
+  return false
+
+proc initMessageReactionItem*(emojiId: EmojiId): MessageReactionItem =
+  result.emojiId = emojiId
+
+proc `$`*(self: MessageReactionItem): string =
+  var reactions = ""
+  for r in self.reactions:
+    reactions = reactions & "displayName: " & r.displayName &  "  publicKey: " & r.publicKey & "  reactionId: " & 
+    r.reactionId & "\n"
+
+  result = fmt"""MessageReactionItem(
+    emojiId: {self.emojiId},
+    amIReactedWithThisEmoji: {self.amIReactedWithThisEmoji},
+    reactionsCount: {self.reactions.len},
+    reactions: {reactions}
+    ]"""
+
+proc emojiId*(self: MessageReactionItem): EmojiId {.inline.} = 
+  self.emojiId
+
+proc amIReactedWithThisEmoji*(self: MessageReactionItem): bool {.inline.} = 
+  self.amIReactedWithThisEmoji
+
+proc numberOfReactions*(self: MessageReactionItem): int {.inline.} = 
+  self.reactions.len
+
+proc jsonArrayOfUsersReactedWithThisEmoji*(self: MessageReactionItem): JsonNode {.inline.} = 
+  var users: seq[string]
+  for r in self.reactions:
+    users.add(r.displayName)
+  return %* users
+
+proc shouldAddReaction*(self: MessageReactionItem, userPublicKey: string): bool = 
+  for r in self.reactions:
+    if (r.publicKey == userPublicKey):
+      return false
+  return true
+
+proc getReactionId*(self: MessageReactionItem, userPublicKey: string): string = 
+  for r in self.reactions:
+    if (r.publicKey == userPublicKey):
+      return r.reactionId
+  return ""
+
+proc addReaction*(self: var MessageReactionItem, amIReactedWithThisEmoji: bool, userPublicKey: string, 
```

Comment:
Same here, but I'll stop commenting on each one. Just change them using `ctrl+shift+f` if you feel like it